.github/workflows/gradle.yml

```diff
@@ -15,21 +15,43 @@ jobs:
       - name: Check images exist
         run: ./tools/bin/check_images_exist.sh
 
-      - name: Cache java deps
+      - name: Docker Caching
         uses: actions/cache@v2
         with:
-          path: ~/.gradle
-          key: gradle-${{ hashFiles('**/*.gradle') }}
+          path: |
+            /tmp/docker-registry
+          key: docker-${{ runner.os }}-${{ hashFiles('Dockerfile') }}-${{ github.sha }}
```

Comment:
I'm using Dockerfile hashes even though the contents of the image might change, because this is only used to speed up builds, not to decide if it should skip a Docker build.

Reply:
Yes, it is intended to store the cache entries with different keys. However, restore-keys allows it to read from different keys. It makes it easier to benchmark re-running builds for Docker with almost no downside (except maybe more cache evictions since it isn't overwriting keys as much).